Remove CMake find_library call for librt

rizsotto opened the pull request:
CMake `find_library` calls always resolves the library as shared object
over the static library. This hurts static linking and cross compilations.
(The installed `abslTargets.cmake` file had the host's librt.so file with
absolute path.)

This fix just adds the librt as a link option to the base library, without
resolving the location of the library.

derekmauro commented:
This causes our CMake build on MacOS to fail with this error message:

```
/Applications/Xcode_11.3.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -target x86_64-apple-macos10.14 -isysroot /Applications/Xcode_11.3.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk -L/tmpfs/src/Debug.AhRxa7RO/bin/Debug -F/tmpfs/src/Debug.AhRxa7RO/bin/Debug -filelist /tmpfs/src/Debug.AhRxa7RO/absl/base/absl.build/Debug/absl_strerror_test.build/Objects-normal/x86_64/absl_strerror_test.LinkFileList -Xlinker -object_path_lto -Xlinker /tmpfs/src/Debug.AhRxa7RO/absl/base/absl.build/Debug/absl_strerror_test.build/Objects-normal/x86_64/absl_strerror_test_lto.o -Xlinker -no_deduplicate -Wl,-search_paths_first -Wl,-headerpad_max_install_names /tmpfs/src/Debug.AhRxa7RO/absl/base/Debug/libabsl_strerror.a /tmpfs/src/Debug.AhRxa7RO/absl/strings/Debug/libabsl_strings.a /tmpfs/src/Debug.AhRxa7RO/lib/Debug/libgmockd.a /tmpfs/src/Debug.AhRxa7RO/lib/Debug/libgtest_maind.a /tmpfs/src/Debug.AhRxa7RO/absl/strings/Debug/libabsl_strings_internal.a /tmpfs/src/Debug.AhRxa7RO/absl/base/Debug/libabsl_base.a /tmpfs/src/Debug.AhRxa7RO/absl/base/Debug/libabsl_dynamic_annotations.a /tmpfs/src/Debug.AhRxa7RO/absl/base/Debug/libabsl_spinlock_wait.a -lrt /tmpfs/src/Debug.AhRxa7RO/absl/numeric/Debug/libabsl_int128.a /tmpfs/src/Debug.AhRxa7RO/absl/base/Debug/libabsl_throw_delegate.a /tmpfs/src/Debug.AhRxa7RO/absl/base/Debug/libabsl_raw_logging_internal.a /tmpfs/src/Debug.AhRxa7RO/absl/base/Debug/libabsl_log_severity.a /tmpfs/src/Debug.AhRxa7RO/lib/Debug/libgtestd.a -Xlinker -dependency_info -Xlinker /tmpfs/src/Debug.AhRxa7RO/absl/base/absl.build/Debug/absl_strerror_test.build/Objects-normal/x86_64/absl_strerror_test_dependency_info.dat -o /tmpfs/src/Debug.AhRxa7RO/bin/Debug/absl_strerror_test
ld: library not found for -lrt
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

rizsotto committed:
Remove CMake find_library call for librt

CMake `find_library` calls always resolves the library as shared object
over the static library. This hurts static linking and cross compilations.
(The installed `abslTargets.cmake` file had the host's librt file with
absolute path.)

This fix just adds the librt as a link option to the base library, whithout
resolving the location of the library.

rizsotto commented:
Thanks for the message. I've updated the PR to leave the resolution and the conditional check there, but the assignment of the library is not the value itself, but the `-lrt` literal. Hope that will fix the OSX build. (By the way, how can I look to the CI build results?)

derekmauro commented:
Thanks, this seems like a reasonable solution.

I think I have to change some setting to allow you to see the build results. I'll have to look for it.
